Sources/Kitura/RouteRegex.swift

```diff
@@ -52,10 +55,17 @@ public class RouteRegex {
     ///
     /// - Parameter pattern: Optional string
     /// - Parameter allowPartialMatch: True if a partial match is allowed. Defaults to false.
-    /// - Returns: A tuple of the compiled `RegularExpressionType?` and array of keys
-    internal func buildRegex(fromPattern: String?, allowPartialMatch: Bool = false) -> (RegularExpressionType?, [String]?) {
+    /// - Returns: A tuple of the compiled `RegularExpressionType?`, a bool as to whether or not
+    ///            this is a simple String compare, and array of keys
+    internal func buildRegex(fromPattern: String?, allowPartialMatch: Bool = false) -> (RegularExpressionType?, Bool, [String]?) {
         guard let pattern = fromPattern else {
-            return (nil, nil)
+            return (nil, false, nil)
+        }
+        
+        // Check and see if the pattern is a simple string (no captures and not a regular expression)
+        let range = NSMakeRange(0, pattern.characters.count)
+        guard let _ = simpleStringChecker.firstMatch(in: pattern, options: [], range: range) else {
```

Comment:
The following can be achieved by
```swift
let complexRouteCharacters = CharacterSet(charactersIn: "*.:+?()[]\")

if pattern.rangeOfCharacter(from: complexRouteCharacters) != nil {
  return (nil, true, nil)
} 
```
Maybe we should measure which method is faster.